hera_cal/lstbin_simple.py

```diff
@@ -0,0 +1,1677 @@
+"""
+An attempt at a simpler LST binner that makes more assumptions but runs faster.
+
+In particular, we assume that all baselines have the same time array and frequency array,
+and that each is present throughout the data array. This allows a vectorization.
+"""
+from __future__ import annotations
+
+import numpy as np
+from . import utils
+import warnings
+from pathlib import Path
+from .lstbin import config_lst_bin_files, sigma_clip
+from . import abscal
+import os
+from . import io
+import logging
+from hera_qm.metrics_io import read_a_priori_ant_flags
+from . import apply_cal
+from typing import Sequence
+import argparse
+from pyuvdata.uvdata.uvh5 import FastUVH5Meta
+from pyuvdata import utils as uvutils
+from .red_groups import RedundantGroups
+import h5py
+from functools import partial
+import yaml
+
+try:
+    profile
+except NameError:
+    def profile(fnc):
+        return fnc
+
+logger = logging.getLogger(__name__)
+
+@profile
+def simple_lst_bin(
+    data: np.ndarray,
+    data_lsts: np.ndarray,
+    baselines: list[tuple[int, int]],
+    lst_bin_edges: np.ndarray,
```

Comment:
this doesn't specify whether this is left or right edges. it seems like it would be easier for the user to provide bin centers (as is the protocol for pyuvdata and UVData objects) and then when you got to run `np.digitize` you shift by width/2 to feed it bin edges. lmk what you think

Reply:
These are *all* the edges, i.e. the bins are *assumed* to be contiguous, but not necessarily all the same size. The output number of bins will be `len(lst_bin_edges) - 1`. I will clarify this in the docstring. I think this is the most reasonable way to input the bins, as it is more general than just sending in bin-centres, and (IMHO) more easy to reason about.

Reply:
These are *all* the edges, i.e. the bins are *assumed* to be contiguous, but not necessarily all the same size. The output number of bins will be `len(lst_bin_edges) - 1`. I will clarify this in the docstring. I think this is the most reasonable way to input the bins, as it is more general than just sending in bin-centres, and (IMHO) more easy to reason about.